File: backend/backend/src/main/java/com/bezkoder/springjwt/security/services/ClusterService.java
package com.bezkoder.springjwt.security.services;

import com.bezkoder.springjwt.models.*;

import java.util.List;

public interface ClusterService {
    ClusterStatistics getClusterStatistics();
    ClusterHealth getClusterHealth();
    List<Cluster> getAllClusters();
    Cluster getClusterById(Long id);
    Cluster createCluster(Cluster cluster);
    Cluster updateCluster(Long id, Cluster clusterDetails);
    void deleteCluster(Long id);
    Cluster addServerToCluster(Long id, Server serverDetails);
    Cluster createOrUpdateMaintenanceSettings(Long clusterId, MaintenanceSettings maintenanceSettings); // Nouvelle méthode
    Cluster addOrUpdateRepositories(Long maintenanceSettingsId, List<Repository> repositories); // Nouvelle méthode

}

File: frontend/font-end/main/src/app/services/cluster.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Cluster } from '../models/Cluster';
import { MaintenanceSettings } from '../models/MaintenanceSettings';
import { Repository } from '../models/Repository';

@Injectable({
  providedIn: 'root'
})
export class ClusterService {
  private apiUrl = 'http://localhost:8081/api/clusters';

  constructor(private http: HttpClient) { }

  getAllClusters(): Observable<Cluster[]> {
    return this.http.get<Cluster[]>(this.apiUrl);
  }

  getClusterById(id: number): Observable<Cluster> {
    return this.http.get<Cluster>(`${this.apiUrl}/${id}`);
  }

  createCluster(cluster: Cluster): Observable<Cluster> {
    return this.http.post<Cluster>(this.apiUrl, cluster);
  }

  updateCluster(id: number, cluster: Cluster): Observable<Cluster> {
    return this.http.put<Cluster>(`${this.apiUrl}/${id}`, cluster);
  }

  deleteCluster(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  createOrUpdateMaintenanceSettings(clusterId: number, maintenanceSettings: MaintenanceSettings): Observable<Cluster> {
    return this.http.post<Cluster>(`${this.apiUrl}/${clusterId}/maintenance-settings`, maintenanceSettings);
  }

  addOrUpdateRepositories(maintenanceSettingsId: number, repositories: Repository[]): Observable<Cluster> {
    return this.http.post<Cluster>(`${this.apiUrl}/maintenance-settings/${maintenanceSettingsId}/repositories`, repositories);
  }
  getClusterStatistics(): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/statistics`);
  }

}
